File: May 2022/May_07.java
class Solution {
    public boolean find132pattern(int[] nums) {
        int n = nums.length;
        int[] min = new int[n];
        min[0] = nums[0];
        for(int i = 1; i < n; i++)
            min[i] = Math.min(min[i - 1], nums[i]);
        
        Stack<Integer> st = new Stack<>();
        for(int i = n - 1; i > 0; i--){
            while(!st.isEmpty() && st.peek() <= min[i])
                st.pop();
            if(!st.isEmpty() && nums[i] > st.peek())
                return true;
            st.add(nums[i]);
        }
        return false;
    }
} 

File: May 2022/May_07.cpp
class Solution {
public:
    bool find132pattern(vector<int>& nums) {
        int n = nums.size();
        int mini[n];
        mini[0] = nums[0];
        for(int i = 1; i < n; i++)
            mini[i] = min(mini[i - 1], nums[i]);
        stack<int> st;
        for(int i = n - 1; i > 0; i--){
            while(!st.empty() && st.top() <= mini[i])
                st.pop();
            if(!st.empty() && st.top() < nums[i])
                return true;
            st.push(nums[i]);
        }
        return false;
    }
};
